include/fc/io/cfile.hpp

```diff
@@ -94,14 +119,24 @@ class cfile {
    }
 
    void write( const char* d, size_t n ) {
-      size_t result = fwrite( d, 1, n, _file.get() );
+      const size_t WRITE_SIZE = 1;
+
+      if(!is_open())
+         throw std::ios_base::failure("cfile is not open");
+
+      size_t result = fwrite( d, WRITE_SIZE, n, _file.get() );
       if( result != n ) {
          throw std::ios_base::failure( "cfile: " + _file_path.generic_string() +
                                        " unable to write " + std::to_string( n ) + " bytes; only wrote " + std::to_string( result ) );
       }
+      // Keep up with the size  of the file.
+      _size +=  (WRITE_SIZE * n);
```

Comment:
Do we have a restriction that only one cfile can read/write to a file? I don't think there is that restriction before this change.

Reply:
so what you are saying  is that it  might have been possible for  two instances of cfile to open the same file at the  same time, before these  changes?

Reply:
so what you are saying  is that it  might have been possible for  two instances of cfile to open the same file at the  same time, before these  changes?

Reply:
Yes. I don't think there was anything to prevent that use case.

Reply:
ok. that is a good  point. Is this a requirement? Who do I talk to about  this?

Reply:
maybe remaining is  only a requirement for `cfile_datastream`?

Reply:
Can't you use the impl you had? 
https://www.geeksforgeeks.org/eof-and-feof-in-c/
https://www.tutorialspoint.com/c_standard_library/c_function_feof.htm

```
    bool eof() const  {
        if(!_open)
            throw std::ios_base::failure("cfile is not open");

        if(feof(_file.get()))return true;

        int c = fgetc(_file.get());
        if(EOF == c)return true;

        return (EOF == ungetc(c, _file.get()));
    }
```

Reply:
good point.